topics/ui/index.md

```diff
@@ -0,0 +1,10 @@
+---
+aliases: user-interface
+created_by: Glenda Schroeder
+display_name: User interface
+released: '1985'
+short_description: A UI is what users use to control a program through a graphical interface
+topic: ui
+wikipedia_url: https://en.wikipedia.org/wiki/User_interface
+---
+The goal of user interface design is to produce a user interface which makes it easy, efficient, and enjoyable (user-friendly) to operate a machine in the way which produces the desired result (i.e. maximum usability). This generally means that the operator needs to provide minimal input to achieve the desired output, and also that the machine minimizes undesired outputs to the user.
```

Comment:
This description's subject seems like it might be for "User Interface Design" instead of "User Interface". Should this topic addition be `ui-design`?

Reply:
This comment still seems relevant, any thoughts to changing this topic to `ui-design`?

Reply:
> This comment still seems relevant, any thoughts to changing this topic to `ui-design`?

I'm not really sure how I feel. I feel like `ui-design` is like a part of `ui` if that makes any sense?

Reply:
The conflict I see is that we're targeting UI in all the other places except the long description, see here for the wikipedia article on UI design, which is different than just UI: https://en.wikipedia.org/wiki/User_interface_design

Reply:
ah i see, then i say we change it to `ui-design` and i'll make another, more generalized pr about `ui` later on.